Improve: Auto-recover from stuck password masking during login

mpconley opened the pull request:
#### Brief overview of PR changes/additions

Adds a safety timeout that automatically clears password masking if it gets stuck during the login phase. The timeout only applies during the first 5 minutes of a connection and respects the user's password masking preference.

#### Motivation for adding to Mudlet

Some users on slow connections experience password masking getting stuck after login, requiring them to disable the feature entirely. This happens when the game server fails to properly signal the end of password entry due to network issues. This change provides automatic recovery while preserving the security benefit of password masking.

#### Other info (issues closed, discussion etc)

Reported by @zookaongit - intermittent issue on slow connections where password masking doesn't clear after login.

mpconley committed:
Add safety timeout for password masking during login phase

When a MUD server sends IAC WILL ECHO to enable password mode, some servers
may fail to send IAC WONT ECHO due to network issues or bugs, leaving the
command line stuck in password masking mode.

This adds a 60-second safety timeout that automatically clears password
masking if the server never sends WONT ECHO. The timeout only activates
during the first 5 minutes of a connection (login phase) to avoid affecting
legitimate in-game password prompts (e.g., admin commands).

The timeout is skipped if password masking is disabled in preferences.

vadi2 commented on src/ctelnet.cpp:
```diff
@@ -2661,6 +2668,30 @@ void cTelnet::processTelnetCommand(const std::string& telnetCommand)
+                    // password prompts later in the session (e.g., admin commands).
+                    // Skip this if the user has disabled password masking entirely.
+                    constexpr qint64 LOGIN_PHASE_MS = 5 * 60 * 1000;  // 5 minutes
+                    constexpr int PASSWORD_TIMEOUT_MS = 60 * 1000;    // 60 seconds
```
We can use modern std::chrono literals for this, as we use C++20.

mpconley replied:
Done!

mpconley committed:
Apply vadi2's review: Use modern C++20 chrono literals

- Replace manual millisecond calculations with chrono literals (5min, 60s)
- Update submodules to latest versions
